### ISSUE TITLE
Crahes on server shutdown

### ISSUE THREAD

**Issue by Binero** [gnembon/fabric-carpet#1192]
The latest 1.18 version gives the following error, preventing a proper server shut down:

```
[19:19:18] [Server thread/ERROR]: Exception stopping the server
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_2168.method_9259(int)" because "source" is null
at carpet.CarpetSettings$CarpetPermissionLevel.validate(CarpetSettings.java:132) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at carpet.CarpetSettings$CarpetPermissionLevel.validate(CarpetSettings.java:130) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at carpet.settings.ParsedRule.set(ParsedRule.java:161) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at carpet.settings.ParsedRule.resetToDefault(ParsedRule.java:235) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at carpet.settings.SettingsManager.detachServer(SettingsManager.java:138) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at carpet.CarpetServer.onServerDoneClosing(CarpetServer.java:199) ~[fabric-carpet-1.18-1.4.55+v211130.jar:?]
at net.minecraft.server.MinecraftServer.handler$zdg000$serverDoneClosed(MinecraftServer.java:2275) ~[intermediary-server-1.18.jar:?]
at net.minecraft.server.MinecraftServer.method_3782(MinecraftServer.java:630) ~[intermediary-server-1.18.jar:?]
at net.minecraft.class_3176.method_3782(class_3176.java:567) ~[intermediary-server-1.18.jar:?]
at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:723) ~[intermediary-server-1.18.jar:?]
at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:270) ~[intermediary-server-1.18.jar:?]
at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by altrisi** [comment]
There's already been a fixed version released, get 1.4.56

**Comment by Binero** [comment]
That was fast indeed! Cheers!
